### PR TITLE
singletons now use modern c++ implementation

### PR DESCRIPTION
**Pull request by prince-chrismc**
I tested these in my own

### CODE REVIEW

**Comment by prince-chrismc** [comment]
basic window still launches

File: NeverGreenGrove/NeverGreenGrove/src/NeverGreenGrove.cpp [hunk 1]
```diff
@@ -28,33 +28,38 @@ SOFTWARE.
...
-GlfwWindow* MainWindow = nullptr;
-
 int main()
 {
-   std::cout << "Welcome to Template!" << std::endl;
+   std::cout << "Welcome to Never Green Grove!" << std::endl;
 
    // Create a GLFWwindow
-   GlfwWindow* window = GlfwWindowFactory::GetInstance().CreateNewWindow("Template by <Author>", GlfwWindow::DEFAULT_WIDTH, GlfwWindow::DEFAULT_HEIGHT);
+   std::shared_ptr<GlfwWindow> window = GlfwWindowFactory::GetInstance()->CreateNewWindow("Never Green Grove - Prepare to get lost!");
```
Comment (prince-chrismc) [inline NeverGreenGrove/NeverGreenGrove/src/NeverGreenGrove.cpp:42]: please evaluate!

**Review by DanielPri — APPROVED** [review]
Perhaps clarify changes in person

**Review by DJNGattuso — COMMENTED** [review]
Similar to @DanielPri, maybe a quick explanation about certain things will be good. I also don't have the chance right now to run the code and see if its working. 

I'll approve it after

**Comment by prince-chrismc** [comment]
I would love to explain it, I just really learnt those implementing this. Explaining will help solidify my understanding

**Comment by DJNGattuso** [comment]
Great! So maybe we can find a good time this week for it

**Pull request merged**